packages/framework/aqueduct/src/container-runtime-factories/baseContainerRuntimeFactory.ts

```diff
@@ -97,18 +101,64 @@ export class BaseContainerRuntimeFactory
 		this.registry = new FluidDataStoreRegistry(this.registryEntries);
 	}
 
-	public async instantiateFirstTime(runtime: ContainerRuntime): Promise<void> {
+	/**
+	 * @deprecated Use the overload taking in an IContainerRuntime
+	 */
+	// eslint-disable-next-line import/no-deprecated -- ContainerRuntime class to be moved to internal scope
+	public async instantiateFirstTime(runtime: ContainerRuntime): Promise<void>;
+	/**
+	 * Called the one time the container is created, and not on any subsequent load.
+	 * i.e. only when it's initialized on the client that first created it
+	 * @param runtime - The runtime for the container being initialized
+	 */
+	public async instantiateFirstTime(runtime: IContainerRuntime): Promise<void>;
+	public async instantiateFirstTime(
+		// eslint-disable-next-line import/no-deprecated -- ContainerRuntime class to be moved to internal scope
+		runtime: IContainerRuntime | ContainerRuntime,
+	): Promise<void> {
 		await this.containerInitializingFirstTime(runtime);
 		await this.containerHasInitialized(runtime);
 	}
 
-	public async instantiateFromExisting(runtime: ContainerRuntime): Promise<void> {
+	/**
+	 * @deprecated Use the overload taking in an IContainerRuntime
+	 */
+	// eslint-disable-next-line import/no-deprecated -- ContainerRuntime class to be moved to internal scope
+	public async instantiateFromExisting(runtime: ContainerRuntime): Promise<void>;
```

Comment:
There doesn't appear to be any need for this. `ContainerRuntime` implements `IContainerRuntime`. It is non-breaking to accept a wider type.

Reply:
Wouldn't it be breaking for someone extending the class and expecting the narrower type?

Reply:
Hmm but I guess "interface | class" type doesn't really help this. In practice though nothing here is going to change before CR is just gone, so seems ok to pass on it.

Reply:
Sealed is possible - we add `@sealed`

Reply:
Does that actually mean anything?  On class functions?

Reply:
It doesn't do anything functionally; it just documents our support contract. If something is `@sealed`, and a user extends / overrides it, then we may break them with future changes. 

That said, newly marking something as `@sealed` is itself a breaking change, so I think we would still need to use the same migration approach here. We would just have more freedom to change the new (sealed) methods in the future.